awx/main/fields.py

```diff
@@ -790,13 +790,11 @@ def schema(self, model_instance):
                 'extra_vars': {
                     'type': 'object',
                     'patternProperties': {
-                        # http://docs.ansible.com/ansible/playbooks_variables.html#what-makes-a-valid-variable-name
-                        '^[a-zA-Z_]+[a-zA-Z0-9_]*$': {'type': 'string'},
+                        r'^(?:(?:{(?:{|%)[^{}]*?(?:%|})})|(?:[a-zA-Z_]+[a-zA-Z0-9_]*)+)+$': {"anyOf": [{'type': 'string'}, {'$ref': '#/properties/extra_vars'}]}
```

Comment:
This regex is ridiculous, but the only alternative I can see is to just make it `r'^.+$'`